.cloudbuild/scripts/internal/git/knownhosts.go

```diff
@@ -0,0 +1,42 @@
+package git
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/gravitational/trace"
+)
+
+func touchTempFile() (string, error) {
+	tempFile, err := ioutil.TempFile("", "*")
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating known hosts file")
+	}
+	defer tempFile.Close()
+
+	err = tempFile.Chmod(0600)
+	if err != nil {
+		return "", trace.Wrap(err, "failed securing known hosts file")
+	}
+
+	return tempFile.Name(), nil
+}
+
+func configureKnownHosts() (string, error) {
+	knownHostsFile, err := touchTempFile()
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating known_hosts file")
+	}
+
+	script := fmt.Sprintf("ssh-keyscan -H github.com > %q 2>/dev/null", knownHostsFile)
```

Comment:
Could we use https://docs.github.com/en/rest/reference/meta#get-github-meta-information for this instead? Looks like it's just a single https fetch and a json field access

Reply:
Believe it or not, that's not actually the tricky part. The tricky part is formatting it into a `known_hosts` file that the OpenSSH client will reliably accept. I tried manually constructing one using the [knownhosts](https://pkg.go.dev/golang.org/x/crypto/ssh/knownhosts) package, and it only seemed to work about 50% of the time.

Reply:
Isn't it just those lines in `ssh_keys` each prefixed with `github.com `?

Reply:
Granted, my testing of this was a little frazzled by trying to get _something_ that worked up and going, and after spending too long on my preferred approach (i.e. doing all this with `go-git`), I may have gone a bit overboard on the "just shell out to it" approach...

Reply:
I stand corrected :smiley:! Polling the metadata service is nicer, cleainer and more secure. Works like an absolute charrm!

I expect that I just didn't have all the keys I needed in the file when testing before.

+10 to Gryffindor.